src/RadixDlt.NetworkGateway.PostgresIntegration/LedgerExtension/PostgresLedgerExtenderService.cs

```diff
@@ -1048,6 +1072,42 @@ private async Task<ExtendLedgerReport> ProcessTransactions(ReadWriteDbContext db
                                 IsLocked = methodRoyaltyEntry.IsLocked,
                             });
                         }
+
+                        if (substateData is CoreModel.TypeInfoModuleFieldTypeInfoSubstate typeInfoSubstate)
+                        {
+                            if (typeInfoSubstate.Value.Details is CoreModel.ObjectTypeInfoDetails { InstanceSchema: not null } objectTypeInfoDetails)
+                            {
+                                if (objectTypeInfoDetails.InstanceSchema.ProvidedTypes.Count != 1)
+                                {
+                                    throw new NotSupportedException("Expected non fungible data with only one data type entry.");
+                                }
+
+                                nonFungibleDataSchemaHistoryToAdd.Add(new NonFungibleDataSchemaHistory
```

Comment:
Open question: shouldn't we use "non fungible schema history" (without this "data") to follow existing pattern?